### ISSUE TITLE
Fix Style/MethodCallWithArgsParentheses issue with anonymous splat forwarding

### ISSUE THREAD

**Issue by bkuhlmann** [rubocop/rubocop#11418]
## Expected behavior

Hello. :wave: I'm logging this as a new issue since I incorrectly reported this in #11417.

When using the following configuration:

``` yaml
Style/MethodCallWithArgsParentheses:
  Enabled: true
  EnforcedStyle: omit_parentheses
```

I'd like to Rubocop to not flag single (`*`) or double (`**`) splats as violations since the use of parenthesis is needed in these situations.

## Actual behavior

I'm seeing behavior where using parenthesis is flagged by RuboCop to drop the parenthesis.

## Steps to reproduce the problem

Use the following code:

``` ruby
class Demo
  def initialize(handler: Object.new, **)
    super(**)
    @handler = handler
  end
end
```

When you use RuboCop to analyze the above, you'll end up with the following errors:

```
Offenses:

/Users/bkuhlmann/Engineering/Misc/snippet:17:10: C: [Correctable] Style/MethodCallWithArgsParentheses: Omit parentheses for method calls with arguments. (https://rubystyle.guide#method-invocation-parens)
    super(**)
         ^^^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
```

## RuboCop version

```
1.42.0 (using Parser 3.1.3.0, rubocop-ast 1.24.1, running on ruby 3.2.0) [arm64-darwin22.2.0]
  - rubocop-performance 1.15.2
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.16.0
  - rubocop-thread_safety 0.4.4
```
